client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py

```diff
@@ -18,35 +21,42 @@ def parse_s3_url(url: str) -> (str, str):
     return parsed.netloc, parsed.path
 
 
-def download_https(url: str, dest_path: Optional[str] = None, with_progress: bool = True):
+def download_https(
+    url: str,
+    dest_path: Optional[str] = None,
+    with_progress: bool = True,
+):
     dest_path = get_destination_path(url, dest_path)
     fetch_request = requests.get(url, stream=True)
     total_size = int(fetch_request.headers["content-length"])
     block_size = 1024 * 512
-    print(f"Downloading {url} to {dest_path}")
+    logger.info("Downloading %s to %s", url, dest_path)
     with tqdm(
         total=total_size,
         unit="iB",
         unit_scale=True,
         disable=(not with_progress),
-    ) as progress_bar:
-        with open(dest_path, "wb") as f:
-            for data in fetch_request.iter_content(block_size):
-                progress_bar.update(len(data))
-                f.write(data)
+    ) as progress_bar, open(dest_path, "wb") as f:
+        for data in fetch_request.iter_content(block_size):
+            progress_bar.update(len(data))
+            f.write(data)
 
 
-def get_destination_path(url: str, dest_path: Optional[str], recursive_from_prefix: Optional[str] = None) -> str:
+def get_destination_path(
+    url: str,
+    dest_path: Optional[str],
+    recursive_from_prefix: Optional[str] = None,
+) -> str:
     if not dest_path:
         dest_path = os.getcwd()
     dest_path = os.path.abspath(dest_path)
 
-    if not recursive_from_prefix:
-        # If the dest path is a directory, append the url's file name to it.
-        if os.path.isdir(dest_path):
-            return os.path.join(dest_path, os.path.basename(url))
     if not os.path.isdir(dest_path):
```

Comment:
Looks like this might be more than linting changes. This probably needs the latest commits to be pulled in?

Reply:
yes, these aren't automatic changes, these were ones I made myself after receiving the following linting error:
```python
client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py:47:5: SIM102 Use a single `if` statement instead of nested `if` statements
```

I basically just moved around the logic but the functionality _should_ stay the same. Does it look to you like the functionality will be the same or is there another way you think it should be written? Of course this would be easy to verify if we had unit tests...